src/de.rs

```diff
@@ -169,7 +169,16 @@ impl<'de, 'a, R: dec::Read<'de>> serde::Deserializer<'de> for &'a mut Deserializ
         }
         match dec::if_major(byte) {
             major::UNSIGNED => de.deserialize_u64(visitor),
-            major::NEGATIVE => de.deserialize_i64(visitor),
+            major::NEGATIVE => {
+                // CBOR supports negative integers up to -2^64 which is less than i64::MIN. Only
+                // treat it as i128, if it is outside the i64 range.
+                let value = i128::decode(&mut de.reader)?;
+                if value >= i64::MIN as i128 {
+                    visitor.visit_i64(value as i64)
+                } else {
+                    visitor.visit_i128(value)
+                }
```

Comment:
```suggestion
               match value.try_into() {
                    Ok(value) => visitor.visit_i64(value as i64),
                    Err(TryFromIntError(value)) => visitor.visit_i128(value),
                }
```

(or not, it just means we don't need the cast & check)

Reply:
Oh, that's nice!